### ISSUE TITLE
Problem at Storage::file();

### ISSUE THREAD

**Issue by uibar** [laravel/framework#9572]
I cannot get any files from this function if I specify a directory even though the dir name is correct. I can see the files if I use allFiles() bun I need only the files in the current dir to be diplayed.

I need to say that this worked before today as I've seen an update to the filesystem drive. Most certein todays update broke the method.

```php
$files = Storage::files('seeders');
dd($files);

// returns empty array
```

**Comment by ArjanSchouten** [comment]
Which update are you referring to? Which versions? There is no ```file()```/```files()``` method in [Filesystemmanager](https://github.com/laravel/framework/blob/5925ed9bae03c86ea0d6902568ca0ffd159d7a61/src/Illuminate/Filesystem/FilesystemManager.php)

**Comment by GrahamCampbell** [comment]
Please ask on the forums. :)

**Comment by basementjack** [comment]
Graham, can we get this opened back up?

This functionality is documented on the 5.1 documentation at http://laravel.com/docs/5.1/filesystem#directories

After doing some research, it looks like it only will retrieve files from approot\storage\app and it's subdirectories.

This is in contrast to the http://laravel.com/docs/5.1/filesystem#storing-files  which presently appears to take any valid path on the file system.

It's either an issue that should be addressed in the files command, or the documentation should be updated so dev's know that it's limited to the storage/app directory and it's children. Thanks.
- Jack

**Comment by loranger** [comment]
+1 It's a real documentation or Storage issue.

**Comment by GrahamCampbell** [comment]
https://github.com/laravel/framework/blob/5.1/src/Illuminate/Filesystem/FilesystemAdapter.php#L221

**Comment by GrahamCampbell** [comment]
You should pass the whole directory path here.

**Comment by taylorotwell** [comment]
Can we get real confirmation on this instead of locking the conversation?

**Comment by taylorotwell** [comment]
Relevant documentation: 

> When using the local driver, note that all file operations are relative to the root directory defined in your configuration file. By default, this value is set to the storage/app directory. Therefore, the following method would store a file in storage/app/file.txt:

`Storage::disk('local')->put('file.txt', 'Contents');`

**Comment by taylorotwell** [comment]
http://laravel.com/docs/5.1/filesystem

**Comment by loranger** [comment]
It's a real documentation __reader__ issue :)

The documentation permit us to get to the essential part, allowing us to bypass the mandatory.
Laravel is so intuitive that I assumed I was able to browse my whole file system (or at least my whole project tree) as I would do with a native php methods wrapper and it leads @basementjack and I to a misunderstanding empty array without any error (obviously).

Maybe the doc could remind us that we use a specific disk and driver but it may quickly escalate to an idiotic situation such as "of course, do not forget to open you php tags"...
Maybe the documentation could teach us, impatient developers, to take enough time to check the fundamental but, honestly, good luck with that.

Today I learned my experience does not allow me to ignore the “getting started” part and I had to disturb @GrahamCampbell and @taylorotwell in order to discover this. So sorry about this.
I wish we won't be that numerous to use that way before understanding Storage class, and I'd like to use this useless issue comment in order to thank you guys for the Laravel framework.

Sorry for the noise.
